admin/spec/routes/school.spec.js

```diff
@@ -60,21 +66,48 @@ describe('school controller:', () => {
 
     beforeEach(() => {
       sandbox = sinon.sandbox.create()
+      next = jasmine.createSpy('next')
     })
 
     afterEach(() => {
       sandbox.restore()
     })
 
-    it('displays the generate pins list page', async (done) => {
-      const res = getRes()
-      const req = getReq(goodReqParams)
-      const controller = require('../../controllers/school').getGeneratePinsList
-      spyOn(res, 'render').and.returnValue(null)
-      await controller(req, res)
-      expect(res.locals.pageTitle).toBe('Select pupils')
-      expect(res.render).toHaveBeenCalled()
-      done()
+    describe('when the school is found in the database', () => {
+      beforeEach(() => {
+        sandbox.mock(School).expects('findOne').chain('exec').resolves(new School({ name: 'Test School' }))
+        controller = proxyquire('../../controllers/school.js', {
+          '../models/school': School
+        }).getGeneratePinsList
+      })
+
+      it('displays the generate pins list page', async (done) => {
+        const res = getRes()
+        const req = getReq(goodReqParams)
+        spyOn(generatePinsService, 'getPupils').and.returnValue(Promise.resolve({}))
+        spyOn(res, 'render').and.returnValue(null)
+        await controller(req, res, next)
+        expect(res.locals.pageTitle).toBe('Select pupils')
+        expect(res.render).toHaveBeenCalled()
+        done()
+      })
+    })
+
+    describe('when the school is not found in the database', () => {
+      beforeEach(() => {
+        sandbox.mock(School).expects('findOne').chain('exec').resolves(null)
+        controller = proxyquire('../../controllers/school.js', {
+          '../models/school': School
+        }).getGeneratePinsList
+      })
+      it('it throws an error', async (done) => {
+        const res = getRes()
+        const req = getReq(goodReqParams)
+        await controller(req, res, next)
+        expect(next).toHaveBeenCalled()
+        expect(res.statusCode).toBe(200)
```

Comment:
should you be looking for status 200?

Reply:
that expectation is removed since it doesn't make sense.